indexer/src/fastnear_indexer.rs

```diff
@@ -0,0 +1,301 @@
+        publish_sender: &Sender<PublishData>,
+        addresses_to_rollup_ids: &HashMap<AccountId, u32>,
+    ) -> Result<(), Error> {
+        debug!(target: "fastnear_indexer", "Parsing block: {:?}", block.block.header.height);
+        for shard in block.shards {
+            for receipt_execution_outcome in shard.receipt_execution_outcomes {
+                let receiver_id = &receipt_execution_outcome.receipt.receiver_id;
+                debug!(target: "fastnear_indexer", "Processing receipt for receiver_id: {}", receiver_id);
+                if let Some(rollup_id) = addresses_to_rollup_ids.get(receiver_id) {
+                    trace!(target: "fastnear_indexer", "Processing receipt for rollup_id: {}", rollup_id);
+                    if !Self::is_successful_execution(&receipt_execution_outcome) {
+                        trace!(target: "fastnear_indexer", "Skipping unsuccessful execution for rollup_id: {}", rollup_id);
+                        continue;
+                    }
+
+                    let partial_candidate_data = Self::receipt_filter_map(
+                        receipt_execution_outcome.receipt.receipt,
+                        *rollup_id
+                    );
+
+                    if let (Some(partial_data), Some(tx_hash)) = (partial_candidate_data, receipt_execution_outcome.tx_hash) {
+                        let candidate_data = PartialCandidateDataWithBlockTxHash {
+                            rollup_id: *rollup_id,
+                            payloads: partial_data.payloads,
+                            tx_hash,
+                            block_hash: block.block.header.hash,
+                        };
+                        debug!(target: "fastnear_indexer", "Sending candidate data for rollup_id: {}", rollup_id);
+                        Self::send(&candidate_data, publish_sender).await?;
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn stream_latest_blocks(&self) -> mpsc::Receiver<BlockWithTxHashes> {
+        info!(target: "fastnear_indexer", "Starting block stream");
+        let (block_sender, block_receiver) = mpsc::channel(100);
+        let client = self.client.clone();
+
+        tokio::spawn(async move {
+            loop {
+                match Self::fetch_latest_block(&client).await {
+                    Ok(block) => {
+                        if block_sender.send(block.clone()).await.is_err() {
+                            error!(target: "fastnear_indexer", "Failed to send block to channel");
+                            break;
+                        }
+                        info!(target: "fastnear_indexer", "Successfully fetched and sent latest block with id: {}", block.block.header.height);
+                    }
+                    Err(e) => error!(target: "fastnear_indexer", "Error fetching latest block: {:?}", e),
+                }
+                tokio::time::sleep(std::time::Duration::from_secs(1)).await;
```

Comment:
magic number - does this introuce latency? is there a way to make it non arbitrary?

Reply:
It is question to you actually, I didn't touch this piece of code :)

As a rule, industry uses exponential backoff for such purposes. I will introduce it too

Reply:
yeah me putting it in a draft PR doesn't mean its correct is it ;)

Reply:
@firatsertgoz I re-read your question. What do you mean by 

>  is there a way to make it non arbitrary?

It is non-arbitrary already :)